Device Update Agent v1.2 for Ubuntu Core

Nox-MSFT opened the pull request:
- Port Ubuntu Core Snap support from feature/snap-1 branch
- Update an Agent version to 1.2.0
- Disable diagnostic and command helper components
- Fix shelcheck errors and warnings in example-snap-update.sh

Nox-MSFT committed:
- Port Ubuntu Core Snap support from feature/snap-1 branch
- Update an Agent version to 1.2.0
- Disable diagnostic and command helper components
- Fix shelcheck errors and warnings in example-snap-update.sh

jw-msft commented on CMakeLists.txt:
```diff
@@ -95,7 +95,7 @@ include (aduc_helpers)
                                "The major part of the semantic version")
 
-set_cache_with_env_or_default (ADUC_VERSION_MINOR "1" STRING
+set_cache_with_env_or_default (ADUC_VERSION_MINOR "2" STRING
```
should this minor ver only change in release/1.2 branch so it does not rev until merged back into develop for the commit id that would be tagged as 1.2 release?

jw-msft commented on snap/hooks/configure:
```diff
@@ -0,0 +1,173 @@
+
+        # Create home dir
+        if [ ! -d "$adu_home_dir" ]; then
+            mkdir -p "$adu_home_dir"
```
was shellcheck run on all scripts?

jw-msft commented on CMakeLists.txt:
```diff
@@ -212,6 +212,46 @@ set (
+if (ADUC_BUILD_UNIT_TESTS)
+    set (
+        EIS_DATA_PATH
+        "/tmp/eis_conn_info.txt"
```
release artifacts are often built with -u so this would have the wrong path in production binaries.  Instead, can have 'testdata' folder under ut folder and the copy_test_data CMake macro will be run that will copy it under a deterministic test directory (under tmp):

https://github.com/Azure/iot-hub-device-update/blob/99ab06fbdf9dc91f4015eb83fc2f4a2bf977346c/cmake/aduc_helpers.cmake#L51

jw-msft reviewed (commented):
some comments for a couple of things that should be improved

jw-msft committed:
fix crash when fmt specifier exists in scriptOutput str (#730)

Nox-MSFT committed:
Add required do-cmake-option for Ubuntu Core snap build
Update deviceupdate-agent snap version to 1.2.0
Update example du-config.json (for snap)
Change version number of the example update for snap

Nox-MSFT committed:
Merge branch 'develop' into feature/snap-v1.2
